sdks/python/apache_beam/dataframe/doctests.py

```diff
@@ -290,34 +298,60 @@ def to_callable(cond):
     super(BeamDataframeDoctestRunner, self).__init__(
         checker=_DeferrredDataframeOutputChecker(self._test_env, use_beam),
         **kwargs)
+    self.skipped = 0
+    self.wont_implement = 0
 
   def run(self, test, **kwargs):
     self._checker.reset()
-    if test.name in self._skip:
-      for example in test.examples:
-        if any(should_skip(example) for should_skip in self._skip[test.name]):
-          example.source = 'pass'
-          example.want = ''
     for example in test.examples:
-      if example.exc_msg is None:
+      if any(should_skip(example)
+             for should_skip in self._skip.get(test.name, [])):
+        example.source = 'pass'
+        example.want = ''
+        self.skipped += 1
+      elif example.exc_msg is None and any(
+          wont_implement(example)
+          for wont_implement in self._wont_implement.get(test.name, [])):
         # Don't fail doctests that raise this error.
         example.exc_msg = (
             'apache_beam.dataframe.frame_base.WontImplementError: ...')
+        self.wont_implement += 1
     with self._test_env.context():
-      return super(BeamDataframeDoctestRunner, self).run(test, **kwargs)
+      result = super(BeamDataframeDoctestRunner, self).run(test, **kwargs)
+      return result
 
   def fake_pandas_module(self):
     return self._test_env.fake_pandas_module()
 
+  def summarize(self):
+    super(BeamDataframeDoctestRunner, self).summarize()
+    if self.failures:
+      return
+    total_test_cases = self.skipped + self.tries
+    will_implement = total_test_cases - self.wont_implement
+    print("%d total test cases." % total_test_cases)
+    print(
+        "%d will implement, %d won't implement." %
```

Comment:
Just because they're not won't implement, doesn't mean we will (e.g .they could be skipped, or they're already implemented in which case the future tense is odd). Maybe break this down as 

```
Xxx total test cases.
    Xxx skipped
    Xxx won't implement
        Yyy reason A
        Yyy reason B
        ...
    Xxx passed

```